src/entryPoints/AuthModule/TwoFaAuth/TwoFaLanding.res

```diff
@@ -0,0 +1,126 @@
+      <p className={`${p2Regular} text-jp-gray-700`}>
+        {"or "->React.string}
+        <span
+          className="cursor-pointer underline underline-offset-2 text-blue-600"
+          onClick={_ => {
+            setTwoFaStatus(_ => TwoFaNotExpired)
+            setTwoFaPageState(_ => TOTP_INPUT_RECOVERY_CODE)
+          }}>
+          {"Use recovery-code"->React.string}
+        </span>
+      </p>
+    | RC_ATTEMPTS_EXPIRED =>
+      <p className={`${p2Regular} text-jp-gray-700`}>
+        {"or "->React.string}
+        <span
+          className="cursor-pointer underline underline-offset-2 text-blue-600"
+          onClick={_ => {
+            setTwoFaStatus(_ => TwoFaNotExpired)
+            setTwoFaPageState(_ => TOTP_SHOW_QR)
+          }}>
+          {"Use totp"->React.string}
+        </span>
+      </p>
+    | TWO_FA_EXPIRED => React.null
+    }
+
+    <BackgroundImageWrapper>
+      <div className="h-full w-full flex flex-col gap-4 items-center justify-center p-6 ">
+        <div
+          className="bg-white px-6 py-12 rounded-md border w-1/3 text-center font-semibold flex flex-col gap-4">
+          <p>
+            {"There have been multiple unsuccessful sign-in attempts for this account. Please wait a moment before trying again."->React.string}
+          </p>
+          {belowComponent}
+        </div>
+        <div className="text-grey-200 flex gap-2">
+          {"Log in with a different account?"->React.string}
+          <p
+            className="underline cursor-pointer underline-offset-2 hover:text-blue-700"
+            onClick={_ => setAuthStatus(LoggedOut)}>
+            {"Click here to log out."->React.string}
+          </p>
+        </div>
+      </div>
+    </BackgroundImageWrapper>
+  }
+}
+
+@react.component
+let make = () => {
+  open TwoFaTypes
+  let getURL = APIUtils.useGetURL()
+  let fetchDetails = APIUtils.useGetMethod()
+  let {setAuthStatus} = React.useContext(AuthInfoProvider.authStatusContext)
+  let (twoFaStatus, setTwoFaStatus) = React.useState(_ => TwoFaNotExpired)
+  let (twoFaPageState, setTwoFaPageState) = React.useState(_ => TOTP_SHOW_QR)
+  let (screenState, setScreenState) = React.useState(_ => PageLoaderWrapper.Loading)
+
+  let handlePageBasedOnAttempts = responseDict => {
+    switch responseDict {
+    | Some(value) =>
+      if value.totp.attemptsRemaining > 0 && value.recoveryCode.attemptsRemaining > 0 {
+        setTwoFaStatus(_ => TwoFaNotExpired)
+        setTwoFaPageState(_ => TOTP_SHOW_QR)
+      } else if value.totp.attemptsRemaining == 0 && value.recoveryCode.attemptsRemaining == 0 {
+        setTwoFaStatus(_ => TwoFaExpired(TWO_FA_EXPIRED))
+      } else if value.totp.attemptsRemaining == 0 {
+        setTwoFaStatus(_ => TwoFaExpired(TOTP_ATTEMPTS_EXPIRED))
+      } else if value.recoveryCode.attemptsRemaining == 0 {
+        setTwoFaStatus(_ => TwoFaExpired(RC_ATTEMPTS_EXPIRED))
+      }
+    | None => setTwoFaStatus(_ => TwoFaNotExpired)
+    }
+  }
```

Comment:
```suggestion
  let handlePageBasedOnAttempts = responseDict => {
    let updateStatus = (totpAttemptsRemaining, recoveryAttemptsRemaining) => {
      switch (totpAttemptsRemaining, recoveryAttemptsRemaining) {
      | (0, 0) => setTwoFaStatus(_ => TwoFaExpired(TWO_FA_EXPIRED))
      | (0, _) => setTwoFaStatus(_ => TwoFaExpired(TOTP_ATTEMPTS_EXPIRED))
      | (_, 0) => setTwoFaStatus(_ => TwoFaExpired(RC_ATTEMPTS_EXPIRED))
      | _ => {
          setTwoFaStatus(_ => TwoFaNotExpired)
          setTwoFaPageState(_ => TOTP_SHOW_QR)
        }
      }
    }

    switch responseDict {
    | Some(value) =>
      updateStatus(value.totp.attemptsRemaining, value.recoveryCode.attemptsRemaining)
    | None => setTwoFaStatus(_ => TwoFaNotExpired)
    }
  }
```

Reply:
with enums it would make sense to use switch for the above case the code is bit clear with the if statement only.